.github/workflows/check-backport-changelog.yml

```diff
@@ -0,0 +1,58 @@
+name: Verify Core Backport Changlog
+
+on:
+    pull_request:
+        types: [opened, synchronize]
+        paths:
+            - 'lib/**'
+            - '!lib/experimental/**'
+            - 'packages/**/*.php'
+            - 'phpunit/**'
+jobs:
+    check:
+        name: Check CHANGELOG diff
+        runs-on: ubuntu-latest
+        steps:
+            - name: 'Get PR commit count'
+              run: echo "PR_COMMIT_COUNT=$(( ${{ github.event.pull_request.commits }} + 1 ))" >> "${GITHUB_ENV}"
+            - name: Checkout code
+              uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+              with:
+                  ref: ${{ github.event.pull_request.head.ref }}
+                  repository: ${{ github.event.pull_request.head.repo.full_name }}
+                  fetch-depth: ${{ env.PR_COMMIT_COUNT }}
+                  show-progress: ${{ runner.debug == '1' && 'true' || 'false' }}
+            - name: 'Fetch relevant history from origin'
+              run: git fetch origin ${{ github.event.pull_request.base.ref }}
+            - name: Check CHANGELOG status
+              env:
+                  PR_NUMBER: ${{ github.event.number }}
+              run: |
+                  changelog_folder="backport-changelog"
+
+                  # Fail if the PR doesn't touch the changelog
+                  if git diff --quiet ${{ github.event.pull_request.base.sha }} HEAD -- "$changelog_folder"; then
+                    echo "Please create a core backport PR and add a file with the name <core-pr-number>.md in the $changelog_folder folder with the core backport PR URL and a list item with this PR URL."
+                    echo "If changes are related to an existing, open core PR, you may add this PR URL to the core PR's file."
+                    exit 1
+                  fi
+
+                  # Find any changelog file that contains the Gutenberg PR link
+                  gutenberg_pr_url="https://github\.com/WordPress/gutenberg/pull/${PR_NUMBER}"
+                  changelog_file=$(grep -rl "[-*] ${gutenberg_pr_url}" "${changelog_folder}" | head -n 1)
+
+                  # Confirm that there is an entry containing the Gutenberg PR link
+                  if [[ -z "${changelog_file}" ]]; then
+                    echo "Please create a core backport PR and add a file with the name <core-pr-number>.md in the $changelog_folder folder with the core backport PR URL and a list item with this PR URL."
```

Comment:
> with the name <core-pr-number>.md

Why not use `${changelog_file}` to say exactly what the file's name should be?

Reply:
We don't know the core PR number in advance